### ISSUE TITLE
Syntax error: Unexpected token in src/index.ts

### ISSUE THREAD

**Issue by articobandurini** [react-native-device-info/react-native-device-info#1167]
During compilation i get this error
```
error: bundling failed: SyntaxError: .../node_modules/react-native-device-info/src/index.ts: Unexpected token (12:12)
```
I guess this is maybe because I am not using TypeScript in my project. Should I?
I tried to remove the field `react-native` from the `package.json` of your module - and this error gone.
Could you please recommend how to fix this?

**Comment by mikehardy** [comment]
There is no requirement to use typescript in order to use react-native-device-info, even the module itself is implemented in typescript.
We compile it to javascript prior to distributing it on npmjs

If you can reproduce this failure with the example, we can investigate, but it is likely something project specific about how you are consuming the module, perhaps in some nonstandard way like not from npmjs.com ?

**Comment by vignesarul** [comment]
I have the same error today after upgrading to 8.0.1 from 2.x.

I had to remove Line 5```react-native": "src/index.ts",``` from node_modules/react-native-device-info/package.json to make it work.

**Comment by articobandurini** [comment]
> If you can reproduce this failure with the example, we can investigate, but it is likely something project specific about how you are consuming the module, perhaps in some nonstandard way like not from npmjs.com ?

I installed the module in usual way using `npm install --save react-native-device-info`.
For some reason, when `"react-native": "src/index.ts"` is present in `node_modules/react-native-device-info/package.json` - the issue is happening, builder tries to load `src/index.ts` and fails. Without this field everything works well.
Manually removing a field from `package.json` is not a sustainable solution...

**Comment by mikehardy** [comment]
No, manual edits of package.json is not sustainable at all (even with the glorious patch-package)
However my comment stands: If you can reproduce this failure with the example, we can investigate

With no reproduction, I consider this not actionable.

**Comment by articobandurini** [comment]
Sorry, but what example you mean?
@vignesarul also encountered the issue

**Comment by articobandurini** [comment]
Could you please also shed some light on why the field "react-native" is in the package.json, and why it points to TypeScript version of the module? This can help us a lot to overcome the issue.

**Comment by mikehardy** [comment]
I mean the example bundled with the module, like most react-native modules. For the avoidance of doubt, this: https://github.com/react-native-device-info/react-native-device-info/tree/master/example

Check out the module and run the example.
If it works, something is likely wrong with your project *or* perhaps we have some subtle expectation being violated, but at least we'll have *you* running the example and reproducing on that so *I* can actually see a failure.

To date it hasn't failed for me at all

To your second question, you can see the build process by examining package.json. Researching that build process should lead you to the answer

**Comment by articobandurini** [comment]
I have to admit that the example works as expected.
Still I have the same syntax error when `"react-native": "src/index.ts"` is present and I do assembleRelease.

**Comment by mikehardy** [comment]
If it doesn't show up either in the example or with some simple change to the example (so I can reproduce it), then I *can't* reproduce it so I can't help, sorry

**Comment by articobandurini** [comment]
Hello again!
After setting up bitbucket pipelines for building the project, I again stumbled upon the issue.
The solution is to use "sed" command to remove the line containing "index.ts" pattern after "npm install":
`sed -i '/index.ts/d' node_modules/react-native-device-info/package.json`
Maybe it will be helpful to somebody. The issue is really weird, all other react-native modules work well.

**Comment by mikehardy** [comment]
@articobandurini so if you had a setup that reproduced this problem perhaps we could finally figure it out? I've never reproduced it so it hasn't been actionable for me. If you could put a github repo up that showed the problem, we could advance this

**Comment by articobandurini** [comment]
Would it be okay if I share the content of the package.json file?

**Comment by mikehardy** [comment]
Post a minimal reproducible example here on github. That's the gold standard for getting things done, with regard to reproducing and fixing problems. I can then clone it and see it fail. "minimal" is emphasized

**Comment by dquessenberry** [comment]
I'm not sure why this is closed as I don't see a fix or workaround and the issue definitely exists for some. I upgraded to v8.0.2, but I also got the same error on v8.0.1 too. I downgraded to v7.3.1 and seems to work ok for me. BTW, I am currently running on the following:

react: 16.9.0
react-native: 0.61.5

The app is never able to start after upgrading to v8.0.2. I am not using typescript in my app. Unfortunately, I do not have a project on GitHub that is publicly available. I am planning to upgrade to the latest react native soon, so hopefully that resolves it.

**Comment by mikehardy** [comment]
@dquessenberry the comment directly above your confusion explains things:

> Post a minimal reproducible example here on github. That's the gold standard for getting things done, with regard to reproducing and fixing problems. I can then clone it and see it fail. "minimal" is emphasized

Do that. I see reproduction of bug in module? I fix problem.

No reproduction. No attention at all.

**Comment by articobandurini** [comment]
Wow, that was unexpectedly pretty impolite answer.
The bug we are talking about was introduced by this commit: https://github.com/react-native-device-info/react-native-device-info/commit/8af015f832fd5a4125d4b97979b86555fb710e64#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80
The thing is all about importing and exporting types.
Downgrading the module version to 8.0.0 fixes the issue.
It is advisable to pay more attention to what other participants of open-source community report, without unneccesary requirements to post anything on github in order to prove that your module actually has bugs.
As for "gold standard" - don't you think that such a breaking change deserves more that a minor version upgrade?
That's a ritoric question by the way. Good luck!

**Comment by antoinefBam** [comment]
I don't know if it may help, but I had a similar `tsc` issue after upgrading this library. After some digging I also identified [this particular commit](https://github.com/react-native-device-info/react-native-device-info/commit/8af015f832fd5a4125d4b97979b86555fb710e64#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80)

It was actually a problem with the typescript version. Indeed `export type` and `import type` statements are only supported from [version 3.8 of TypeScript](https://devblogs.microsoft.com/typescript/announcing-typescript-3-8/)

Upgrading fixed the issue 👍

**Comment by mikehardy** [comment]
@antoinefBam that explains it and moves things forward, thank you. I can add documentation around this. Too late for the breaking change alas.

**Comment by SahilPrabhakar7** [comment]
> I have the same error today after upgrading to 8.0.1 from 2.x.
> 
> I had to remove Line 5`react-native": "src/index.ts",` from node_modules/react-native-device-info/package.json to make it work.

Thanks  it worked

**Comment by thilinaa** [comment]
I had a similar issue. I upgraded it from 2.1.2 to the 8.4.4 version and my react-native version is  (0.59.10).
I don't use typescript in my project.
Having the following dev dependencies fixed this issue for me.

 "devDependencies": {
         "@babel/core": "^7.9.0",
         "@babel/runtime": "^7.9.2",
         //other dev dependencies
    }

Found this solution from the accepted answer from this, https://stackoverflow.com/questions/62270438/react-native-error-bundling-failed-syntaxerror-react-native-projects-appone-n

**Comment by thilinaa** [comment]
Hey @mikehardy, Yeah upgrading the react-native is on our road map. But I have to wait till January to do that.
BTW, I'm using the Xcode version 13.1 and it's working after a few modifications in the Podfile.

**Comment by taj567** [comment]
> I have the same error today after upgrading to 8.0.1 from 2.x.
> 
> I had to remove Line 5`react-native": "src/index.ts",` from node_modules/react-native-device-info/package.json to make it work.

it worked thanks.

**Comment by ChenhuaWANG22** [comment]
> I had a similar issue. I upgraded it from 2.1.2 to the 8.4.4 version and my react-native version is (0.59.10). I don't use typescript in my project. Having the following dev dependencies fixed this issue for me.
> 
> "devDependencies": { "@babel/core": "^7.9.0", "@babel/runtime": "^7.9.2", //other dev dependencies }
> 
> Found this solution from the accepted answer from this, https://stackoverflow.com/questions/62270438/react-native-error-bundling-failed-syntaxerror-react-native-projects-appone-n

@mikehardy 
Better to add some info like below in the change log for people who do not use typescript:
https://github.com/react-native-device-info/react-native-device-info/commit/c50b1b93621e6241331e5422c28ec774845a304a

"
if you don't use typescript, Babel added support type-only imports/exports through this [merge request](https://github.com/babel/babel/pull/11171) and is available from version 7.9.0.

so update your babel version as shown below:
 "devDependencies": {
         "@babel/core": "^7.9.0",
         "@babel/runtime": "^7.9.2",
         //other dev dependencies
    } 
"
after this sentence:
"Note, if you use typescript, this requires a minimum version of typescript 3.8"

I have tested on my project. The versions were "7.7.4". After updating to "^7.9.0" and  "^7.9.2", it works.
Besides, I have check the example project, they are  "@babel/core": "^7.12.9" and "@babel/runtime": "^7.12.5", which may explain the reason.
https://github.com/react-native-device-info/react-native-device-info/blob/master/example/package.json
